pkg/virt-controller/watch/vm_test.go

```diff
@@ -466,22 +468,298 @@ var _ = Describe("VirtualMachine", func() {
 				vmiInterface.EXPECT().AddVolume(context.Background(), vmi.ObjectMeta.Name, vm.Status.VolumeRequests[0].AddVolumeOptions)
 			}
 
-			vmInterface.EXPECT().Update(context.Background(), gomock.Any(), metav1.UpdateOptions{}).Do(func(ctx context.Context, arg interface{}, options metav1.UpdateOptions) {
-				Expect(arg.(*v1.VirtualMachine).Spec.Template.Spec.Volumes[0].Name).To(Equal("vol1"))
-			}).Return(vm, nil)
-
-			vmInterface.EXPECT().UpdateStatus(context.Background(), gomock.Any(), metav1.UpdateOptions{}).Do(func(ctx context.Context, arg interface{}, options metav1.UpdateOptions) {
-				// vol request shouldn't be cleared until update status observes the new volume change
-				Expect(arg.(*v1.VirtualMachine).Status.VolumeRequests).To(HaveLen(1))
-			}).Return(vm, nil)
-
 			sanityExecute(vm)
+
+			vm, err = virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Get(context.TODO(), vm.Name, metav1.GetOptions{})
+			Expect(err).To(Succeed())
+			Expect(vm.Spec.Template.Spec.Volumes[0].Name).To(Equal("vol1"))
+			Expect(vm.Status.VolumeRequests).To(BeEmpty())
 		},
 
 			Entry("that is running", true),
 			Entry("that is not running", false),
 		)
 
+		Context("add volume request", func() {
+			It("should not be cleared when hotplug fails on Running VM", func() {
+				By("Running VM")
+				vm, vmi := DefaultVirtualMachine(true)
+				vm.Status.Created = true
+				vm.Status.Ready = true
+				vm.Status.VolumeRequests = []v1.VirtualMachineVolumeRequest{
+					{
+						AddVolumeOptions: &v1.AddVolumeOptions{
+							Name:         "vol1",
+							Disk:         &v1.Disk{},
+							VolumeSource: &v1.HotplugVolumeSource{},
+						},
+					},
+				}
+
+				vm, err := virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Create(context.TODO(), vm, metav1.CreateOptions{})
+				Expect(err).To(Succeed())
+				addVirtualMachine(vm)
+				markAsReady(vmi)
```

Comment:
Would it make sense to encapsulate the VM creation in a function? It would reduce some lines and makes the test s more readable

Reply:
Definitely but I am keeping this for future as I want this series to be back-portable.